web-registry/src/pages/Roles/hooks/useRolesSubmit.tsx

```diff
@@ -0,0 +1,205 @@
+                      developer['@type'] === 'regen:Individual'
+                        ? PartyType.User
+                        : PartyType.Organization,
+                    name: developer['schema:name'],
+                    description: developer['schema:description'],
+                    image: developer['schema:image']?.['@value'],
+                    walletId,
+                  },
+                },
+              },
+            });
+            const developerId = res.data?.createParty?.party?.id;
+            projectPatch = { developerId };
+          } else if (doUpdateMetadata) {
+            await updateParty({
+              variables: {
+                input: {
+                  id: developer.id,
+                  partyPatch: {
+                    name: developer['schema:name'],
+                    description: developer['schema:description'],
+                    image: developer['schema:image']?.['@value'],
+                    walletId,
+                  },
+                },
+              },
+            });
+          }
+        }
+
+        const newMetadata = { ...metadata, ...stripIds(values) };
+        // In creation mode, we store all metadata in the off-chain table project.metadata temporarily
+        if (!isEdit)
+          projectPatch = {
+            metadata: newMetadata,
+            ...projectPatch,
+          };
+        // In creation or edit mode, we always store references to the project stakeholders in the project table
+        // which should be in projectPatch if new or updated
+        await updateProject({
+          variables: {
+            input: {
+              id: offChainProject?.id,
+              projectPatch,
+            },
+          },
+        });
+        if (!isEdit) {
+          navigateNext();
+        } else {
+          // In edit mode, we need to update the project on-chain metadata if needed
+          if (doUpdateMetadata) {
+            await projectEditSubmit(newMetadata);
+          }
+        }
+        await metadataReload();
+      } catch (e) {
+        // TODO: Should we display the error banner here?
+        // https://github.com/regen-network/regen-registry/issues/554
+        // console.log(e);
+      }
+    },
+    [
+      offChainProject?.partyByDeveloperId,
+      offChainProject?.id,
+      metadata,
+      isEdit,
+      updateProject,
+      metadataReload,
+      createWallet,
+      updateWallet,
+      createParty,
+      updateParty,
+      navigateNext,
+      projectEditSubmit,
+    ],
+  );
+
+  return { rolesSubmit };
+};
+
+function stripPartyIds(
+  values: ProfileFormValues | undefined,
+): ProfileFormValues | undefined {
+  delete values?.id;
+
+  return values;
+}
+
+function stripIds(values: RolesValues): RolesValues {
+  if (values) {
+    return {
+      'regen:projectDeveloper': stripPartyIds(values['regen:projectDeveloper']),
+    };
+  }
+  return values;
```

Comment:
it looks lke the first `if` block isn’t necessary here, given it’s a required function argument.

Should this be
```suggestion
  if (values['regen:projectDeveloper']) {
    return {
      'regen:projectDeveloper': stripPartyIds(values['regen:projectDeveloper']),
    };
  }
  return values;
```
?